packages/arb-validator-core/ethbridge/arbClient.go

```diff
@@ -190,30 +194,97 @@ type TransactAuth struct {
 	auth *bind.TransactOpts
 }
 
-func (t *TransactAuth) makeTx(ctx context.Context, txFunc func(auth *bind.TransactOpts) (*types.Transaction, error)) (*types.Transaction, error) {
-	const smallNonceRepeatCount = 5
-	const smallNonceError = "Try increasing the gas price or incrementing the nonce."
+func (t *TransactAuth) makeContract(ctx context.Context, contractFunc func(auth *bind.TransactOpts) (ethcommon.Address, *types.Transaction, interface{}, error)) (ethcommon.Address, *types.Transaction, error) {
```

Comment:
Can we unify this with `makeTx`? There's a lot of duplication here

Reply:
done